datafusion/expr-common/src/type_coercion/binary.rs

```diff
@@ -1052,12 +1052,16 @@ fn binary_to_string_coercion(
     match (lhs_type, rhs_type) {
         (Binary, Utf8) => Some(Utf8),
         (Binary, LargeUtf8) => Some(LargeUtf8),
+        (BinaryView, Utf8) => Some(Utf8View),
+        (BinaryView, LargeUtf8) => Some(LargeUtf8),
         (LargeBinary, Utf8) => Some(LargeUtf8),
         (LargeBinary, LargeUtf8) => Some(LargeUtf8),
         (Utf8, Binary) => Some(Utf8),
         (Utf8, LargeBinary) => Some(LargeUtf8),
+        (Utf8, BinaryView) => Some(Utf8View),
         (LargeUtf8, Binary) => Some(LargeUtf8),
         (LargeUtf8, LargeBinary) => Some(LargeUtf8),
+        (LargeUtf8, BinaryView) => Some(LargeUtf8),
```

Comment:
PR https://github.com/apache/datafusion/pull/12092
(LargeUtf8, BinaryView) => Some(Utf8View)

but i think we should coercion to LargeUtf8 ? , since LargeUtf8 is can a huge string. 
(LargeUtf8, BinaryView) => Some(LargeUtf8)

Reply:
I think Arrow actually has the equivalent of `LargeUtf8View` but arrow-rs doesn't support it yet. 

I think the mapping `(LargeUtf8, BinaryView) => Some(LargeUtf8)` makes sense to me for now